### PR TITLE
Virtualize AccordianList to resolve performance issues with many dbs/schemas/tables/fields

### PR DESCRIPTION
**Pull request by tlrobinson**
###### TODO 
- [x] fix layout issues
  - [x] missing divider line between last item of expanded section and header of next section
  - [x] padding at beginning/end of section items
- [x] integrate CellMeasurer into AccordianList to handle overflowing item/section titles
- [x] fix automatic height and popover "constrain to screen" behavior
- [x] Ensure `maxHeight` is propagated to all instances of `<AccordianList>`
  - [x] `DataSelector`
  - [x] `FieldList` nested directly inside `Popover`:
    - [x] `BreakoutWidget`
    - [x] `FieldWidget`
  - [x] `BreakoutPopover`
  - [x] `AggregationPopover`
  - [x] `FilterPopover`

### CODE REVIEW

**Comment by tlrobinson** [comment]
@metabase/core-developers This is almost ready, and it's worth giving a try now.

The only remaining issue I'm aware of is that because `AccordianList` now needs to be provided an explicit height (`react-virtualize` requires it) `Popover`'s `sizeToFit` feature doesn't work. It's kind of a catch 22 because we don't know how much height there is until we pick the `Popover` attachment direction, which we don't do until we know the height of the popover content...

**Comment by tlrobinson** [comment]
@metabase/core-developers Alright, I've fixed the remaining known issues. Please review.

**Comment by salsakran** [comment]
I'm getting a weird height for the popover that is larger than the window height. 
screen shot 2017-11-17 at 5 18 26 pm

When I expand the window, it expands the popover so it's still larger than the window height
screen shot 2017-11-17 at 5 17 57 pm

**Comment by tlrobinson** [comment]
@salsakran Ah yes, I forgot to enable the `maxHeight` functionality on some of the other popovers. Now as long as `AccordianList` or `FieldList` is nested directly inside `Popover` it will work automatically, and I think I've manually handled passing `maxHeight` in the rest of the cases.

**Comment by salsakran** [comment]
seems to work pretty well for me!

File: frontend/src/metabase/components/AccordianList.jsx [hunk 1]
```diff
@@ -56,20 +60,50 @@ export default class AccordianList extends Component {
...
 
+
+    // TODO: reimplement scrolling currently selected item into view
+
     componentDidMount() {
-        // when the component is mounted and an item is selected then scroll to it
-        const element = this.refs.selected && ReactDOM.findDOMNode(this.refs.selected);
-        if (element && element.scrollIntoView && !elementIsInView(element)) {
-            element.scrollIntoView();
+        this._forceUpdateList();
```
Comment (kdoh) [inline frontend/src/metabase/components/AccordianList.jsx]: Would be nice to have a note here about why this is needed.

File: frontend/src/metabase/components/AccordianList.jsx [hunk 2]
```diff
@@ -164,91 +198,151 @@ export default class AccordianList extends Component {
         return this.props.getItemClasses && this.props.getItemClasses(item, itemIndex);
     }
 
+    renderSectionHeader(section, sectionIndex) {
```
Comment (kdoh) [inline frontend/src/metabase/components/AccordianList.jsx]: Would probably be good to remove this if it's not called anywhere (doesn't look like it is).

**Review by kdoh — CHANGES_REQUESTED** [review]
Couple tiny code things we should probably clean up before merging but overall seems reasonable. Dig the addition of examples. 👍 

I'd also *really* love it if we could get at least a couple unit tests onto AccordianList since there's some pretty intense logic going on in there.

File: frontend/src/metabase/components/AccordianList.jsx (hunk 1 above)
Comment (tlrobinson) [reply frontend/src/metabase/components/AccordianList.jsx]: Done.

File: frontend/src/metabase/components/AccordianList.jsx (hunk 2 above)
Comment (tlrobinson) [reply frontend/src/metabase/components/AccordianList.jsx]: Done.

**Comment by tlrobinson** [comment]
@kdoh Thanks, made those changes and added some unit tests.

@metabase/core-developers Anyone else want to review the code?

**Review by kdoh — APPROVED** [review]

**Pull request merged**